SHS-5907: Implementation: Editors can easily set "view" access for Private Pages

codechefmarc opened the pull request:
# READY FOR REVIEW

## Summary
- Adds a new contrib module `content_access_simple` that adds a simpler interface for settings view permissions per node.

## Need Review By (Date)
2024-11-06

## Urgency
low

## Steps to Test
1. Login to the site as an admin
2. Create a Private Page
3. Title it
4. Verify there is a new area below all of the fields called "Access and Permissions"

![Screenshot 2024-11-05 at 2 01 24 PM](https://github.com/user-attachments/assets/56defff3-e5ea-4069-b378-cc24d5652745)

6. Open it up and verify that there is a list of roles under "Visibility" but there is not "Developer", "Anonymous", or "Authenticated".
7. Choose a role, like "Site Manager" and enable view access
8. Save the node
9. Visit the "Access control" tab for this node
10. Verify that under "View any hs_private_page content", the "Site Manager" is checked in addition to "Developer"  and "Authenticated user" (which were the defaults originally but they are hidden as per config)
11. Uncheck "Authenticated user" under view any here, and save
12. Attempt to view the page on a logged in user who is not "Site Manager" role and verify that you cannot access it
13. Attempt to view the page on a logged in user who is "Site Manager" and verify that you can access it
14. Edit the node and verify that under "Access and Permissions" you get a "complex" warning and no roles are listed

![Screenshot 2024-11-05 at 1 59 39 PM](https://github.com/user-attachments/assets/09fe4cfa-15fa-4472-b653-5e7d3684a395)

16. To fix, visit the content type permissions at `/admin/structure/types/manage/hs_private_page/access` and also uncheck "Authenticated user" and save
17. Edit the node again and this time, and this time, mark the node as unpublished and save
18. Edit the node and verify that under "Access and Permissions" you get a "unpublished" warning but the roles are still listed

![Screenshot 2024-11-05 at 1 59 19 PM](https://github.com/user-attachments/assets/49d6b5a3-a429-4239-916b-4c0b8497066d)

20. Visit `/admin/structure/types/manage/hs_private_page/form-display` and drag the "Content Access Simple" "field" to another place in the form and save
21. Edit the node and verify that the Access and Permissions has moved.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208890594589119

Commit:
feat(SHS-5907): WIP testing new content access simple on humsci

Commit:
feat(SHS-5907): Add config for testing content access simple

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-colorful-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-traditional-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://english-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d

Visual Diffs:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-colorful-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-traditional-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://english-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d

Visual Diffs:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d/visualdiffs

Commit:
feat(SHS-5907): Add in specific version of content access simple

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-colorful-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-traditional-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://english-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d

Visual Diffs:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-colorful-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://hs-traditional-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com
- https://english-kq1kvl6usmqwhmiuvq4xo2mfqmdxlysp.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d

Visual Diffs:
- https://dashboard.tugboatqa.com/672a68df21db79eb231fe63d/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-wec8vpwmgnbmhnfeib6xvjvtcbmntwny.tugboatqa.com
- https://hs-colorful-wec8vpwmgnbmhnfeib6xvjvtcbmntwny.tugboatqa.com
- https://hs-traditional-wec8vpwmgnbmhnfeib6xvjvtcbmntwny.tugboatqa.com
- https://english-wec8vpwmgnbmhnfeib6xvjvtcbmntwny.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672a9658f95004fb1cb43a3c

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-colorful-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://english-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672aa2cd8e2485d441d2ba16

dalin- commented on config/default/content_access_simple.settings.yml:
```diff
@@ -0,0 +1,7 @@
+_core:
+  default_config_hash: QfKnBM3eY89y3dqsdVFRwG5E8H8zr3zGJyin6Bi9TJk
+role_config:
+  hidden_roles:
```
We also need to hide: Author, and Intranet Viewer; but we need to show authenticated (and rename it, see elsewhere)

codechefmarc replied:
Fixed.

dalin- reviewed (changes requested):
Phewph, that's a lot to review.  


First up, the the code in the other repo:

----------

https://github.com/codechefmarc/content_access_simple/blob/9b9135ad1b89dcaebe908673c102fa7063fd7deb/content_access_simple.info.yml#L5
We might be able to get a jump and call it 11 compatible?

----------

https://github.com/codechefmarc/content_access_simple/blob/9b9135ad1b89dcaebe908673c102fa7063fd7deb/content_access_simple.module#L46
We also need to check if the extra field has actually been enabled. Something like 

```php
$fieldIsEnabled = $display->getComponent('content_access_simple');
```

----------

https://github.com/codechefmarc/content_access_simple/blob/9b9135ad1b89dcaebe908673c102fa7063fd7deb/src/AccessManager.php#L72

I'm not sure it's a good idea to include "own access" here.  In the spec it talks about mostly ignoring the whole concept (i.e. if you're doing stuff with "own access", then it's no longer "simple").  

-----------

https://github.com/codechefmarc/content_access_simple/blob/9b9135ad1b89dcaebe908673c102fa7063fd7deb/content_access_simple.module#L92

How does Content Access work without our custom module installed?  If you create a new node, then change the defaults for that bundle, does the node use the old defaults, or the new defaults?  I suspect the former (and our new module will do the same), but if it's the latter, then we'll need to update our code to match. 

--------

And then from functional testing:

* Need to rename "Authenticated" as "All logged in users" (just on this form, not globally)
* Need help text "Only the chosen roles will be able to view this content..."  this needs to be configurable.
* Unpublished message doesn't match the spec', and needs to be configurable.
* I'm not sure that your plan for not disabling checkboxes will work.  Site Manager needs to be visible but disabled.

Commit:
Merge branch '11.5.2-release' into SHS-5907--easily-set-view

Commit:
feat(SHS-5907): Update content_access_simple with latest changes for review

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-colorful-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://english-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672aa2cd8e2485d441d2ba16

codechefmarc commented:
@dalin- Updated the code and here are my comments on yours:

1. Drupal 11 - updated the info file and did a check with Upgrade Status module so it is ready for D11
2. Good catch on checking to see if the field is enabled in the default display. I added that feature in.
3. The "own access" here (https://github.com/codechefmarc/content_access_simple/blob/9b9135ad1b89dcaebe908673c102fa7063fd7deb/src/AccessManager.php#L72) is actually our own module permissions, not content access. However, I see your point in wanting to make this module more simple, so I removed it and there is only one permission to use this module now.
4. In content access, they do not show the "Access control" tab when you first create a node. They only show it after the fact on the edit page. The defaults come from the content type settings itself via `content_access_node_access_records` which is a hook that sets permissions on node save. It, in turn calls this line: `return $grants[$node->id()][$op] ?? content_access_get_settings($op, $node->getType());` which essentially gets the per node setting, but if that doesn't exist, use the setting from the content type. Since we're showing our form elements on the add page, we need a way to get the defaults first. Unless you want to follow content access and not show our stuff on the add page and instead only show it on the edit page?
5. Renamed "Authenticated" to "All logged in users"
6. Help text added and is part of the config
7. For the unpublished message - I did make this dynamic and list roles who can view unpublished based on the View Unpublished module permissions which Stanford has, but, not all sites will have that module. So, I can also add a config-based unpublished message that will take precedence if set in config. I like the idea of the dynamic one, but it does sort of depend on the view unpublished module. I'm out of hours for this project this week, so I'll make that change next week.
8. Added logic and config for disabled roles. When I first read the spec, I didn't get it and was going to reach out to you, but I understand now.
9. We can certainly hide Stanford specific roles with their config, and I'll do that in this PR to test next week. As part of the module, I only hid the ones that made sense out of the gate for contrib.

dalin- commented:
@codechefmarc 

> Since we're showing our form elements on the add page, we need a way to get the defaults first. Unless you want to follow content access and not show our stuff on the add page and instead only show it on the edit page?

Makes sense.  No changes needed. 

> For the unpublished message... I'm out of hours for this project this week, so I'll make that change next week.

Sounds good. 

> As part of the module, I only hid the ones that made sense out of the gate for contrib.

Ooooh, smart. 

-----

On
https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com/
I masqueraded as "Lindsey" who is a _Site Manager_
<img width="1038" alt="Screenshot 2024-11-15 at 9 39 05 AM" src="https://github.com/user-attachments/assets/81d95d92-f484-4780-9705-8326ddf7d273">
When I try to edit an existing private page, our new field is not there
<img width="1126" alt="Screenshot 2024-11-15 at 9 39 19 AM" src="https://github.com/user-attachments/assets/c8233ff3-45b3-4e53-b777-d6d102bc128b">

But maybe this is more about generic config vs our config.

--------

Lastly, there's a lot of code changed in this PR that seems unrelated.  Maybe it needs to merge in the source branch.

codechefmarc commented:
Ahh, for "Lindsey", I know why - we have permissions for our module too and that wasn't set as part of this PR. I'll need to make that change for Site Managers in the config and do an update hook as well.

And yup, I had the base incorrect for the merge, so I changed it to the 11.5.2 release branch as the base and now there's only a handful of files changed.

However, I'd like to discuss with you how we do this deploy - if we do want this as a contrib module, once we get everything buttoned up, I'll upload to Drupal.org then change composer.json to point to that version and not my repo.

dalin- commented:
@cienvaras 

> However, I'd like to discuss with you how we do this deploy - if we do want this as a contrib module, once we get everything buttoned up, I'll upload to Drupal.org then change composer.json to point to that version and not my repo.

You might want to bring this up in your weekly call with Albert + SWS.  I'm happy to let you handle it, or let me know if you want my opinion.

cienvaras commented:
@dalin-
> You might want to bring this up in your weekly call with Albert + SWS. I'm happy to let you handle it, or let me know if you want my opinion.

We don't have meeting with SWS, but I can ask Joe on the chat. I'd like your opinion though, I haven't been fully involved on this ticket and there are things I might be missing.

dalin- commented:
@cienvaras 
I suspect that we'll need to go through a couple of rounds on this:

1. Get SWS review on the generic module.
2. Get H&S review of the UI/UX.
3. Publish it to drupal.org
4. Create a follow-up task to integrate the contrib module into our site.

Commit:
feat(SHS-5907): Update code with some bug fixes and changes, add update hook for suhumsci to allow site manager to use content access simple

Commit:
Merge branch '11.6.1-release' into SHS-5907--easily-set-view

Commit:
fix(SHS-5907): Fix linting error

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-colorful-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://english-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://history-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://west-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672aa2cd8e2485d441d2ba16

Commit:
f

Commit:
feat: Switch to Drupal.org content_access_simple version

Commit:
Merge branch '11.6.1-release' into SHS-5907--easily-set-view

Commit:
feat(SHS-5907): Fix content_access_simple settings and revert accidental change from 11.6.1

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-colorful-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://english-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://history-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://west-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672aa2cd8e2485d441d2ba16

codechefmarc commented:
@dalin- and @joegl This is ready for a re-review. The module is now on Drupal.org in a beta state. Tests have been added and I also fixed the issue for `humsci` only to hide intranet viewers and authors.

dalin- committed:
fix: Updated help text messaging to match A.C..

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-colorful-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://english-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://history-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://west-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672aa2cd8e2485d441d2ba16

dalin- reviewed (changes requested):
@codechefmarc 
I did some more testing, and I found some issues:

----------

I made a few tweaks to the help text to match our A.C. rather than use the generic text that will work for most Drupal sites. 

---------

I see that there's a test that fails:
https://github.com/SU-HSDO/suhumsci/actions/runs/12058728785/job/33626009159?pr=1679
but it's something about new sites, and something about Facebook, so I'm assuming that is unrelated to our work. 

-------

I tried testing on Tugboat, but it asploded:
https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com/test-access-control
> MySQL server has gone away

So I can't confirm, but I suspect that the default access control for our content type are not quite right (i.e. /admin/structure/types/manage/hs_private_page/access ).  Can you compare with:
https://app.clickup.com/t/36718269/SHS-5907?block=block-6f9c899d-ca04-48c0-a8ce-1bcecd4b6836

-------

Lastly, the one thing missing is the section of the A.C. about writing an update hook for existing Private Pages to show that Site Managers have access to them:
https://app.clickup.com/t/36718269/SHS-5907?block=block-29cafc21-6b93-4b72-aeb4-197c53ce4d78

codechefmarc commented:
@dalin- 
Thanks for the testing! Some notes on your notes:
- Thanks for updating the unpublished message - though they are using View Unpublished so they will get a dynamic list of roles if that message isn't set. No problem there though.
- Re-ran the tests, that new site one is now passing. That happens sometimes on this repo.
- For Tugboat, I was able to fox the "server has gone away" via: https://docs.tugboatqa.com/troubleshooting/fix-problem-x/#mysql-server-has-gone-away
- Thanks for the reminder - I saw an update hook and did the one to grant access to Content Access Simple, but not to the access permissions to private pages. I will add that.

Commit:
Merge branch '11.6.1-release' into SHS-5907--easily-set-view

Commit:
feat(SHS-5907): Add update hook to grant site managers view access to private pages

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-colorful-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://english-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://history-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://west-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672aa2cd8e2485d441d2ba16

codechefmarc commented:
@dalin- Added the update hook so it will now allow site managers to view all and view own private pages for the defaults to the content type but also updates all existing nodes, if there are changes. If not, they use defaults which were already updated.

dalin- commented:
@codechefmarc 
Thanks for the MySQL fix. 

 I tested and I only found some issues with the error messages:

speling missteak
<img width="1021" alt="Screenshot 2024-12-03 at 12 57 39 PM" src="https://github.com/user-attachments/assets/02d0f6b0-e173-431c-b08b-e7b054faa04b">

odd double-message and HTML encoding issue.  Feel free to switch from `<i>` to `<em>` if that helps. 
<img width="1009" alt="Screenshot 2024-12-03 at 1 02 09 PM" src="https://github.com/user-attachments/assets/7c449710-41a5-4037-9c10-b1ac61eda191">

dalin- reviewed (changes requested):
@codechefmarc 
Just some error message tweaks.

Commit:
Merge branch '11.6.1-release' into SHS-5907--easily-set-view

Commit:
fix(SHS-5907): Update to beta2 of content_access_simple to fix spelling and new handling of messaging

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-colorful-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://hs-traditional-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://english-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://history-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com
- https://west-nibcuwfk1qrcj3cnraoyrgs4360qz81k.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/672aa2cd8e2485d441d2ba16

codechefmarc commented:
@dalin- OK, fixed those three things. In doing so, I found that the messaging could be handled better on the back-end so I reworked it a bit.

dalin- reviewed (commented):
@codechefmarc 
A comment on the contrib module change here:
https://git.drupalcode.org/project/content_access_simple/-/commit/d440e19138012157613361a6ebe39ba8f571c298#note_425015

As for the code changes here, there's a lot going on in `composer.lock`.  I suspect that you used `--with-all-dependencies` but probably should've just done `composer update drupal/content_access_simple`.  But maybe  SWS will be fine with all these changes, I don't know.

Commit:
Merge branch '11.6.1-release' into SHS-5907--easily-set-view

Commit:
chore(SHS-5907): Revert composer.lock, next commit will change content access simple

Commit:
chore(SHS-5907): Update composer.lock with only content access simple update

Commit:
feat(SHS-5907): Update content access simple to 1.0.0-beta3

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://hs-colorful-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://hs-traditional-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://english-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://history-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://west-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6758ee49426df8b73cb48a6c

Visual Diffs:
- https://dashboard.tugboatqa.com/6758ee49426df8b73cb48a6c/visualdiffs

codechefmarc commented:
@dalin- Reverted many of the changes in `composer.lock` except for the `content_access_simple` stuff. Also, in the d.o. repo, thank you for the comments, I did change things to use the `Xss` filter, great idea.

dalin- reviewed (approved):
🚢

cienvaras commented:
@ahughes3 This one is ready for you to review.

cienvaras committed:
Merge branch '11.6.1-release' of github.com:SU-HSDO/suhumsci into SHS-5907--easily-set-view

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://hs-colorful-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://hs-traditional-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://english-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://history-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com
- https://west-vxswjkmrvhbaox6l9wa8hpagurpits5d.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6758ee49426df8b73cb48a6c

Visual Diffs:
- https://dashboard.tugboatqa.com/6758ee49426df8b73cb48a6c/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://hs-colorful-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://hs-traditional-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://english-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://history-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://west-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/676088fd5a118331c6b668c8

ahughes3 commented:
@cienvaras @codechefmarc  @dalin- We are still receiving the MySQL error has gone away message when attempting to save. What steps did you all take when testing to resolve on Tugboat? 

It looks like the pr with recommended fix was merged but maye this line should be added as well to the config file

`- echo "max_allowed_packet=536870912" >> /etc/mysql/conf.d/tugboat.cnf`

Also in the UI we noticed that this image was not the same as the image in the testing results

![image](https://github.com/user-attachments/assets/9ad877ff-b82c-4855-b7d6-dade6c4ff30f)

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://hs-colorful-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://hs-traditional-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://english-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://history-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://west-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/676088fd5a118331c6b668c8

ahughes3 committed:
Updating tugboat config.yml to add additional line for mysql fix

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://hs-colorful-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://hs-traditional-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://english-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://history-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com
- https://west-ibh15omcg5whpusq2merx6dv0zczmp7a.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/676088fd5a118331c6b668c8

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-knqo8qv9ugw2zjo1lh2i33tjz2piayf1.tugboatqa.com
- https://hs-colorful-knqo8qv9ugw2zjo1lh2i33tjz2piayf1.tugboatqa.com
- https://hs-traditional-knqo8qv9ugw2zjo1lh2i33tjz2piayf1.tugboatqa.com
- https://english-knqo8qv9ugw2zjo1lh2i33tjz2piayf1.tugboatqa.com
- https://history-knqo8qv9ugw2zjo1lh2i33tjz2piayf1.tugboatqa.com
- https://west-knqo8qv9ugw2zjo1lh2i33tjz2piayf1.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/676133dd252c7a7b929acbdc

ahughes3 committed:
Update config.yml to ensure packet size persists even if MySQL restarts during the update step

cienvaras committed:
Merge branch '11.6.1-release' of github.com:SU-HSDO/suhumsci into SHS-5907--easily-set-view

cienvaras committed:
fix(shs-5907): update update hook sequence numbers

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-izgdcgisxutmhdmsvawliglg2intakzl.tugboatqa.com
- https://hs-colorful-izgdcgisxutmhdmsvawliglg2intakzl.tugboatqa.com
- https://hs-traditional-izgdcgisxutmhdmsvawliglg2intakzl.tugboatqa.com
- https://english-izgdcgisxutmhdmsvawliglg2intakzl.tugboatqa.com
- https://history-izgdcgisxutmhdmsvawliglg2intakzl.tugboatqa.com
- https://west-izgdcgisxutmhdmsvawliglg2intakzl.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6761d6e1919fa9b3e74ce528

Visual Diffs:
- https://dashboard.tugboatqa.com/6761d6e1919fa9b3e74ce528/visualdiffs

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://hs-colorful-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://hs-traditional-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://english-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://history-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://west-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67630b2e252c7a7b92cff1a0

sethprime reviewed (approved):
Verified that Private Page edit form has "Access and Permissions" area.
Verified that Developer, Anonymous, and Authenticated roles are not displayed in the Access and Permissions area.
Verified that only authorized roles were able to view the node.
Verified that the 'content is unpublished' warning appears on the node edit page.
Verified that the Content Access Simple field can be rearranged in the Private Page form display.

ahughes3 reviewed (approved):
LGTM!

ahughes3 committed:
Merge branch 'develop' into SHS-5907--easily-set-view

ahughes3 commented:
@pookmish @sethprime This is a reminder that this review is at the top of our priority list to get to stage this week

ahughes3 commented:
@codechefmarc we are asking SWS to review this so I want to make sure everything is up to date, I see the there are some tests that failed

codechefmarc commented:
@ahughes3 - I see what's going on, I'll fix and verify, thanks for pointing it out!

Commit:
Merge branch '11.6.3-release' into SHS-5907--easily-set-view

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://hs-colorful-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://hs-traditional-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://english-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://history-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com
- https://west-zq6vphgiftb2pxvim53aktrp2y838boa.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67630b2e252c7a7b92cff1a0

codechefmarc commented:
@ahughes3 - Fixed and ready to go.

joegl commented on config/default/user.role.site_manager.yml:
```diff
@@ -188,6 +188,7 @@ permissions:
   - 'edit terms in hs_person_staff_type'
   - 'edit terms in hs_person_student_type'
   - 'edit terms in hs_publication_type'
+  - 'grant content access simple'
```
What exactly does this permissions allow and could it be renamed to fit into the "access, create, edit, delete, etc.," categories used for other permissions?

codechefmarc replied:
Hi @joegl, welcome back! This permission grants the ability for the site manager to see the new "Access and Permissions" details pane. See screenshot below. I copied the format from the Content Access module which uses `grant content access` as one of their permissions. We can change it if you wish as this module is still not widely used yet (Drupal.org says it's got one install and that may be Stanford?). If we changed it, I would say "access" so it would read `access content access simple` if that works for you. Let me know your preference. Thanks!

![Screenshot 2025-01-27 at 3 33 08 PM](https://github.com/user-attachments/assets/1c6cb35e-62a0-422b-91d6-951768f8a989)

joegl replied:
I agree with @dalin- on changing this to `access content access simple`

joegl reviewed (commented):
@ahughes3 @codechefmarc This looks really neat. I haven't reviewed the custom module code in detail yet, but I'm wondering if we may want to roll this out only to a few select sites first to beta test and work out any kinks instead of push it out to every site. Because it affects access I think it's better to be a little more cautious to start.

Commit:
Merge branch '11.6.4-release' into SHS-5907--easily-set-view

ahughes3 commented:
@joegl If you have a way for us to Beta test this feature on stage and then on prod I'm all for it. 4K did a lot of local testing and we did a good amount of Tugboat testing so we feel good about the functionality but we want to test on stage as soon as possible. 

If you prefer to only have this enabled on select sites these are the sites we would prefer to roll the feature out to as they are waiting on the functionality - [english](https://english.stanford.edu/), [MTL](https://mtl.stanford.edu/), and [Anthropology](https://anthropology.stanford.edu/)

dalin- commented:
I like "access content access simple" rather than "grant content access simple".  

I think the easiest way to roll this out to only a few sites is to remove `su_humsci_profile_update_9722()` which gives the above permission to all 'site_manager' on all sites, and either:
* write a replacement update hook to only add it to these 3 sites
* manually set the permission on these 3 sites after deploy

And create a follow-up ticket to add it everywhere.

joegl commented:
> I like "access content access simple" rather than "grant content access simple".
> 
> I think the easiest way to roll this out to only a few sites is to remove `su_humsci_profile_update_9722()` which gives the above permission to all 'site_manager' on all sites, and either:
> 
> * write a replacement update hook to only add it to these 3 sites
> * manually set the permission on these 3 sites after deploy
> 
> And create a follow-up ticket to add it everywhere.

@dalin- @ahughes3 @codechefmarc 

Either of these works for me, although I think I lean more towards manually enabling it on the 3 sites. We also may need to set up a few temporary `config_ignore` rules so any subsequent deploys to staging don't uninstall the module or wipe out any configuration. I've used a `config_ignore` on a specific module like this before `core.extension:module:content_access_simple`.

With the permission name change and changes to only enable on the 3 sites I'd be ready to merge this and move forward.

Another possible option would be to add an extra "enable" setting into the module itself available on the settings page. That way we could install the module across all sites but only enable it on the 3 sites via the setting (and only have to `config_ignore` the settings page).

dalin- commented:
> We also may need to set up a few temporary `config_ignore` rules...

I wonder if that's more complexity than we really need??? I think we're already ignoring all permissions, so just manually giving the `site_manager` role the `access content access simple` permission on those 3 sites could be enough without adding still more `config_ignore` rules.

joegl commented:
> > We also may need to set up a few temporary `config_ignore` rules...
> 
> I wonder if that's more complexity than we really need??? I think we're already ignoring all permissions, so just manually giving the `site_manager` role the `access content access simple` permission on those 3 sites could be enough without adding still more `config_ignore` rules.

That's a good point. The `--partial` imports we do won't remove permissions or uninstall modules not listed in `core.extension.yml` (which is why we always need to do those in update hooks when we do deprecations). I don't think we need any `config_ignore` settings adding.

joegl commented:
@codechefmarc @dalin- Because the module is in a separate repo and not a PR, how would you like me to go about reviewing the code and offering feedback? I did take a cursory review of the code and didn't see anything noteworthy. I wish the `content_access` module API was more D8+ OOP oriented instead of having to use all those functions but it's unavoidable.

codechefmarc commented:
Since this is a contrib module, probably best to create an issue on d.o so that I can fix there and then give you credit at the same time. :)

Commit:
Merge branch '11.6.4-release' into SHS-5907--easily-set-view

Commit:
chore(SHS-5907): Remove site editors from accessing content access simple

Commit:
chore(SHS-5907): Rename update hook

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-colorful-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-traditional-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://english-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://history-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://west-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7

Visual Diffs:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7/visualdiffs

codechefmarc commented:
@dalin- and @joegl - I've made the following changes:
- In Content Access Simple the permission is now `access content access simple`
- In Humsci repo, I've removed the permission for Site Managers in config and removed the update hook
- I tested on Tugboat and re-added that permission manually to Site Editors and masqueraded as a Site Manager and that worked

Let me know if there are any other changes you need. Thanks!

joegl commented:
> @dalin- and @joegl - I've made the following changes:
> 
> * In Content Access Simple the permission is now `access content access simple`
> * In Humsci repo, I've removed the permission for Site Managers in config and removed the update hook
> * I tested on Tugboat and re-added that permission manually to Site Editors and masqueraded as a Site Manager and that worked
> 
> Let me know if there are any other changes you need. Thanks!

@codechefmarc Looking good, just one thing left I see
* Remove the module from `config/default/core.extension.yml` so it doesn't get installed on all sites. 

@ahughes3 I know we have a prod deploy next Wednesday -- do you still want to try to get this into that release or wait? Either works for me since we're manually activating it on only 3 sites.

Commit:
Merge branch '11.6.4-release' into SHS-5907--easily-set-view

codechefmarc commented:
@joegl - Some guidance here - I attempted to remove this from `core.extension` but it won't work because there is the default config `content_access_simple.settings` that has an error when it attempts to import saying the module isn't installed. We could put this in the `config_ignore.settings`? And then on the three sites we want to test, the config would have to be manually set via drush or some other method (there is no UI to set this config yet).

The other path here is we keep it as is, and because we removed permissions to see it, site managers and non-developers won't see that "Access and Permissions" area. Developers will, of course.

Let me know your thoughts. Thanks!

joegl commented:
> @joegl - Some guidance here - I attempted to remove this from `core.extension` but it won't work because there is the default config `content_access_simple.settings` that has an error when it attempts to import saying the module isn't installed. We could put this in the `config_ignore.settings`? And then on the three sites we want to test, the config would have to be manually set via drush or some other method (there is no UI to set this config yet).
> 
> The other path here is we keep it as is, and because we removed permissions to see it, site managers and non-developers won't see that "Access and Permissions" area. Developers will, of course.
> 
> Let me know your thoughts. Thanks!

I say we go the route of adding the settings to `config_ignore` and set them on the 3 sites when we manually install the module.

I'm pasting the current settings here for reference after they get removed from the PR:
```
role_config:
  hidden_roles:
    - anonymous
    - administrator
    - author
    - intranet_viewer
  disabled_roles:
    - site_manager
debug: false
help_text_view: 'Only the chosen roles will be able to view this content (when published). <i>Site Managers</i> can always view all content.'
unpublished_message: 'These settings only take effect once this page is published. Only <i>Site Managers</i> and <i>Contributors</i> can view unpublished private pages.'
```

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-colorful-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-traditional-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://english-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://history-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://west-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7

Visual Diffs:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7/visualdiffs

Commit:
feat(SHS-5907): Content access simple will not be installed or have default config to prep for limited testing

Commit:
feat(SHS-5907): Remove content access simple settings

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-colorful-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-traditional-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://english-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://history-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://west-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7

Visual Diffs:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7/visualdiffs

codechefmarc commented:
@joegl - OK, removed from `core.extension`. I added `content_access_simple.settings` to the `config_ignore.settings` and tested it locally and it worked fine, but when I pushed to Tugboat, we got errors again. So, I simply removed the settings altogether and we can re-recreate on the 3 sites with what you saved in the previous comment. Build it working just fine now.

ahughes3 commented:
@joegl @cienvaras Any updates or an eta on when this could be tested on Stage?

cienvaras committed:
Merge branch '11.7.1-release' of github.com:SU-HSDO/suhumsci into SHS-5907--easily-set-view

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1679-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-colorful-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://hs-traditional-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://english-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://history-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com
- https://west-cuxprfpktuq9xrkg56bkmghxea5nizx4.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7

Visual Diffs:
- https://dashboard.tugboatqa.com/679ae5cb1728297d1421ccb7/visualdiffs

cienvaras commented:
@ahughes3 @joegl No action needed on our side, all feedback was addressed.

joegl commented:
Thanks everyone, hoping to get this on stage tomorrow 👍

Commit:
Merge branch '11.7.1-release' into SHS-5907--easily-set-view

joegl commented:
@ahughes3 There is additional configuration necessary to get this activated on the 3 sites on the staging environment. Once it's deployed and I've updated those sites I will let you know.

joegl commented:
@ahughes3 @codechefmarc @dalin- This has been deployed and then activated on the 3 sites discussed.
